test/node/object_id_tests.js

```diff
@@ -24,6 +24,192 @@ describe('ObjectId', function () {
     done();
   });
 
+  it('should correctly create ObjectId from ObjectId', function () {
+    var tmp = new ObjectId();
+    expect(new ObjectId(tmp).id).to.deep.equal(Buffer.from(tmp.id, 'hex'));
+  });
+
+  const invalidInputs = [
+    { input: [], description: 'empty array' },
+    { input: ['abcdefŽhijkl'], description: 'nonempty array' },
+    { input: {}, description: 'empty object' }
+  ];
+
+  for (const { input, description } of invalidInputs) {
+    it(`should throw error if ${description} is passed in`, function () {
+      expect(() => new ObjectId(input)).to.throw(TypeError);
+    });
+  }
+
+  it('should throw error if object without an id property is passed in', function () {
+    var tmp = new ObjectId();
+    var objectIdLike = {
+      toHexString: function () {
+        return tmp.toHexString();
+      }
+    };
+
+    expect(() => new ObjectId(objectIdLike)).to.throw(TypeError);
+  });
+
+  it('should correctly create ObjectId from object with valid string id', function () {
+    var objectValidStringId1 = {
```

Comment:
could we name the variables after the valid cases they represent? (something containing `24` and `12` since the length is what matters); similarly for the other string/buffer cases in the file